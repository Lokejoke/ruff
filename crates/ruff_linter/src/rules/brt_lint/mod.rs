//! Rules from [BRT-lint](https://is.muni.cz/auth/osoba/524841).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(Rule::FunctionNamedA, Path::new("BRT001.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("brt_lint").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
