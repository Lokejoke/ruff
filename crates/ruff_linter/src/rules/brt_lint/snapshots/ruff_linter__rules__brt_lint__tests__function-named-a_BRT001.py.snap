---
source: crates/ruff_linter/src/rules/brt_lint/mod.rs
---
BRT001.py:1:5: BRT001 Single lettered function named: `a`
  |
1 | def a(): # BRT001
  |     ^ BRT001
2 |     pass
  |

BRT001.py:4:5: BRT001 Single lettered function named: `A`
  |
2 |     pass
3 | 
4 | def A(): # BRT001
  |     ^ BRT001
5 |     return 
  |
